plugins/release-manager-as-a-service/src/ReleaseManagerAsAService.tsx

```diff
@@ -0,0 +1,179 @@
+import { Alert } from '@material-ui/lab';
+import { CircularProgress, makeStyles } from '@material-ui/core';
+import { useAsync } from 'react-use';
+import React, { useState } from 'react';
+import { useApi, ContentHeader, ErrorBoundary } from '@backstage/core';
+
+import { CreateRc } from './cards/createRc/CreateRc';
+import { getGitHubBatchInfo } from './sideEffects/getGitHubBatchInfo';
+import { Info } from './cards/info/Info';
+import { Patch } from './cards/patchRc/Patch';
+import {
+  ComponentConfigCreateRc,
+  ComponentConfigPatch,
+  ComponentConfigPromoteRc,
+  GhGetBranchResponse,
+  GhGetReleaseResponse,
+  GhGetRepositoryResponse,
+  Project,
+  SetRefetch,
+} from './types/types';
+import { PromoteRc } from './cards/promoteRc/PromoteRc';
+import { releaseManagerAsAServiceApiRef } from './api/serviceApiRef';
+import {
+  ApiClientContext,
+  useApiClientContext,
+} from './components/ProjectContext';
+import { RMaaSApiClient } from './api/RMaaSApiClient';
+
+interface ReleaseManagerAsAServiceProps {
+  project: Project;
+  components?: {
+    default?: {
+      createRc?: ComponentConfigCreateRc;
+      promoteRc?: ComponentConfigPromoteRc;
+      patch?: ComponentConfigPatch;
+    };
+    custom?: ({
+      project,
+      setRefetch,
+      latestRelease,
+      releaseBranch,
+      repository,
+    }: {
+      project: Project;
+      setRefetch: SetRefetch;
+      latestRelease: GhGetReleaseResponse | null;
+      releaseBranch: GhGetBranchResponse | null;
+      repository: GhGetRepositoryResponse;
+    }) => JSX.Element[];
+  };
+}
+
+const useStyles = makeStyles(() => ({
+  root: {
+    maxWidth: '999px',
+  },
+}));
+
+export function ReleaseManagerAsAService({
+  project,
+  components,
+}: ReleaseManagerAsAServiceProps) {
+  const pluginApiClient = useApi(releaseManagerAsAServiceApiRef);
+  const RMaaSApi = new RMaaSApiClient({
+    pluginApiClient,
+    repoPath: `${project.github.org}/${project.github.repo}`,
+  });
+  const classes = useStyles();
+
+  return (
+    <ApiClientContext.Provider value={RMaaSApi}>
+      <div className={classes.root}>
+        <ContentHeader title="Release Manager as a Service (RMaaS™️)" />
+
+        <Cards project={project} components={components} />
+      </div>
+    </ApiClientContext.Provider>
+  );
+}
+
+function Cards({ project, components }: ReleaseManagerAsAServiceProps) {
+  const apiClient = useApiClientContext();
+  const [refetch, setRefetch] = useState(0);
+  const gitHubBatchInfo = useAsync(getGitHubBatchInfo({ apiClient }), [
+    project,
+    refetch,
+  ]);
+
+  if (gitHubBatchInfo.error) {
+    return <Alert severity="error">{gitHubBatchInfo.error.message}</Alert>;
+  }
+
+  if (gitHubBatchInfo.loading) {
+    return (
+      <div style={{ display: 'flex', justifyContent: 'center' }}>
+        <CircularProgress />
+      </div>
+    );
+  }
+
+  if (gitHubBatchInfo.value === undefined) {
+    return <Alert severity="error">Failed to fetch latest GHE release</Alert>;
```

Comment:
Specific reference to GitHub Enterprise; which infers this plugin would not work with GitHub.com ?  if it works with both, suggest any and all references to GHE be generic and instead refer to GitHub as a product (which can be delivered SaaS or onprem with GHE of course).